src/map/skill.cpp

```diff
@@ -12284,30 +12349,44 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 				for (i = 0; i < ARRAYLENGTH(scs); i++)
 					if (tsc->getSCE(scs[i])) status_change_end(bl, scs[i]);
 			}
-			if (hd)
+			if (hd != nullptr)
+			{
 				skill_blockhomun_start(hd, skill_id, skill_get_cooldown(skill_id, skill_lv));
+				if (hd->master != nullptr)
+					skill_blockpc_start(hd->master, skill_id, skill_get_cooldown(skill_id, skill_lv));
+			}
 		}
 		break;
 	case MH_OVERED_BOOST:
-		if (hd && battle_get_master(src)) {
-			sc_start(src, battle_get_master(src), type, 100, skill_lv, skill_get_time(skill_id, skill_lv));
+		if (hd != nullptr) {
```

Comment:
```suggestion
		if (hd != nullptr && hd->master != nullptr) {
```
You've changed the behavior of the skill by removing the master check here. Prior the whole skill would fail if there was no homun + master.